src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/TopicItem.test.jsx

```diff
@@ -0,0 +1,215 @@
+
+    const generalTopicNode = queryByTestId(container, 'course');
+    userEvent.click(queryByLabelText(generalTopicNode, 'Expand'));
+
+    expect(generalTopicNode.querySelector('button[aria-label="Collapse"]')).toBeInTheDocument();
+    expect(generalTopicNode.querySelector('button[aria-label="Expand"]')).not.toBeInTheDocument();
+    expect(generalTopicNode.querySelector('button[aria-label="Delete Topic"]')).not.toBeInTheDocument();
+    expect(generalTopicNode.querySelector('input')).toBeInTheDocument();
+  });
+
+  test('displays expand view of additional discussion topic', async () => {
+    await mockStore(legacyApiResponse);
+    createComponent(additionalTopic);
+
+    const topicCard = queryByTestId(container, '13f106c6-6735-4e84-b097-0456cff55960');
+    userEvent.click(queryByLabelText(topicCard, 'Expand'));
+
+    expect(topicCard.querySelector('button[aria-label="Expand"]')).not.toBeInTheDocument();
+    expect(topicCard.querySelector('button[aria-label="Collapse"]')).toBeInTheDocument();
+    expect(topicCard.querySelector('button[aria-label="Delete Topic"]')).toBeInTheDocument();
+    expect(topicCard.querySelector('input')).toBeInTheDocument();
+  });
+
+  describe('renders delete topic popup', () => {
+    beforeEach(async () => {
+      await mockStore(legacyApiResponse);
+      createComponent(additionalTopic);
+
+      const topicCard = queryByTestId(container, '13f106c6-6735-4e84-b097-0456cff55960');
+      userEvent.click(queryByLabelText(topicCard, 'Expand'));
+      userEvent.click(queryByLabelText(topicCard, 'Delete Topic'));
+    });
+
+    test('when click on Delete topic button', async () => {
+      expect(queryAllByText(container, `${messages.discussionTopicDeletionLabel.defaultMessage}`)).toHaveLength(1);
+    });
+
+    test('with title text', async () => {
+      expect(queryByText(container, `${messages.discussionTopicDeletionLabel.defaultMessage}`)).toBeInTheDocument();
+    });
+
+    test('with helping text', async () => {
+      expect(queryByText(container, `${messages.discussionTopicDeletionHelp.defaultMessage}`)).toBeInTheDocument();
+    });
+
+    test('with a delete button', async () => {
+      expect(queryByText(container, `${messages.deleteButton.defaultMessage}`)).toBeInTheDocument();
+    });
+
+    test('with a cancel button', async () => {
+      expect(queryByText(container, `${messages.cancelButton.defaultMessage}`)).toBeInTheDocument();
+    });
+  });
+
+  test('shows help text on field focus', async () => {
+    await mockStore(legacyApiResponse);
+    createComponent(additionalTopic);
+
+    const topicCard = queryByTestId(container, '13f106c6-6735-4e84-b097-0456cff55960');
+    userEvent.click(queryByLabelText(topicCard, 'Expand'));
+    topicCard.querySelector('input').focus();
+
+    expect(queryByText(topicCard, 'Choose a unique name for your topic')).toBeInTheDocument();
+  });
+
+  test('No errors on field when passing valid topic name', async () => {
```

Comment:
seems like this test belongs to discussiontopics.test.jsx.

Reply:
Yes. After the debugging, I find out this test will never fail because the error array will not initialize if we pass an invalid topic name like a duplicate. so we can move this in legacy form test cases.